concepts/type-conversion/about.md

```diff
@@ -1,10 +1,272 @@
+
+For all other values, `Boolean` returns `true`.
+These values are called _truthy_.
+
+```javascript
+Boolean(-1);
+// => true
+
+Boolean(0);
+// => false
+
+Boolean(' ');
+// => true
+
+Boolean('');
+// => false
+```
+
+Note that because of the described rules, the following values are truthy in JavaScript although they might count as falsy in other languages.
+
+```javascript
+Boolean('0');
+// => true
+
+Boolean('false');
+// => true
+
+Boolean([]);
+// => true
+
+Boolean({});
+// => true
+```
+
+### Converting to a Number
+
+`Number(value)` can be used to convert a value to a number.
+Whitespaces at the beginning and the end of a string are ignored and an empty string is converted to `0`.
+If you try to convert a non-primitive value or a string that does not represent a number, **no** error will be thrown.
+Instead, the result is `NaN` ([Not-A-Number][mdn-nan]).
+
+```javascript
+Number('  -12.34  ');
+// => -12.34
+
+Number('1,2');
+// => NaN
+
+Number('1 2');
+// => NaN
+
+Number('');
+// => 0
+
+Number('10e3');
+// => 10000
+
+Number({ key: '123' });
+// => NaN
+```
+
+Below you can see what `Number` returns for other primitive values.
+
+```javascript
+Number(true);
+// => 1
+
+Number(false);
+// => 0
+
+Number(null);
+// => 0
+
+Number(undefined);
+// => NaN
+```
+
+JavaScript also provides the functions `parseInt` and `parseFloat`.
+They apply laxer rules as to which strings can be converted to a number.
+Because of that, `Number` should be preferred as conversion function to avoid unexpected outcomes.
+
+```javascript
+parseInt('123a45');
+// => 123
+
+Number('123a45');
+// => NaN
+```
+
+### Converting to a String
+
+With `String(value)` you can convert a value to a string.
+The result is what you would expect it to be for primitive values.
+
+```javascript
+String(12.34);
+// => '12.34'
+
+String(false);
+// => 'false'
+
+String(null);
+// => 'null'
+
+String(undefined);
+// => 'undefined'
+```
+
+For arrays, the `String` function will apply the string conversion for each element and join the results with a comma.
+You can also apply the [`join` method][mdn-join] yourself, e.g. to customize the separator.
+However, in these cases `null` and `undefined` will be converted to an empty string.
+
+```javascript
+String([42, null, true, 'abc']);
+// => '42,,true,abc'
+```
+
+For objects, by default `String` returns an unhelpful text.
+
+```javascript
+String({ key: 'value' });
+// => '[object Object]'
+```
+
+You can customize the conversion behavior, e.g. by providing a `toString` method.
+The section "[Object to primitive conversion][custom-conversion]" on javascript.info explains the details.
+
+Another common way to achieve a better string representation for objects and arrays is to use [JSON encoding][json].
+
+```javascript
+const obj = {
+  name: 'Gilda Guerra',
+  address: {
+    city: 'Goiânia',
+  },
+};
+
+JSON.stringify(obj);
+// => '{"name":"Gilda Guerra","address":{"city":"Goiânia"}}'
+```
```

Comment:
Should this text also mention `JSON.parse`?

Reply:
I didn't want to dive too much into this because it will have a separate exercise/concept but I agree that if I show the one direction I should also mention the other way. I added a sentence to cover that.